doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,401 @@
+       +---------+---------+---------+                                 |                |
+       | Subtree | ...     | Subtree |                                 v                |
+       +=========+=========+=========+                          +---------------+       |
+       | Header  | ...     | ...     |                          | OID           |<-----------------+
+       +---------+---------+---------+                          | Num Chunk     |       |          |
+    +--| Entry0  | ...     | ...     |                          | <Offset, len> |       |          |
+    |  +---------+---------+---------+                          | <Offset, len> |-------|----+     |
+    |  | ...     | ...     | ...     |                          | ...           |       |    |     |
+    |  +---------+---------+---------+                          +---------------+       |    |     |
+    |  | EntryK  | ...     | ...     |                                                  |    |     |
+    |  +---------+---------+---------+                                                  |    |     |
+    |                                                                                   |    |     |
+    |                                                                                   |    |     |
+    |  +---------------+  +----------+  +-------------+                                 v    v     |
+    +->| leafnode      |  | leafnode |  | leafnode    |               +------------+------------+  |
+       +===============+  +==========+  +=============+               | Block0     | Block1     |  |
+       | OID           |  | OID      |  | OID         |               +============+============+  |
+       | Block no      |  | Block no |  | Block no    |               | Free       |            |  |
+       | Offset        |  | Offset   |  | Offset      |               +------------+            +  |
+       | Extent        |  | Extent   |  | Extent      |----------+    | Small data | Data       |  |
+       +---------------+  +----------+  +-------------+          |    +------------+            +  |
+                                                                 +-+->| Extent     |            |  |
+                                                                      +------------+------------+  |
+                                                                          |                        |
+                                                                          +------------------------+
+
+
+WAL
+---
+Each AG has a WAL.
+The datas written to the WAL are all metadta updates, free space update and data.
+Note that only data smaller than the predefined threshold needs to be written to the WAL.
+The larger data is written to the unallocated free space and its onode's extent_map is updated accordingly 
+(also on-disk extent map). We statically allocate WAL partition aside from data partition pre-configured.
```

Comment:
"Each AG has a WAL".  This to mean implies that you cannot have a transaction that spans two AGs.  I think this is going to be a problem for you.  Your metadata structures and data structures pertaining to a particular object may be allocated over multiple AGs -- how would you perform an atomic update under those conditions?